department-of-veterans-affairs/caseflow
# Jhunton/appeals 4535

Resolves [APPEALS-4535](https://vajira.max.gov/browse/APPEALS-4535)

### Description
Updated the Hearing Conversion form so that when a VSO user accesses the form to convert a hearing to virtual, they are prompted with permission and acceptance checkboxes that they must click before submitting the form. 

### Acceptance Criteria
- [X] The following affirmations appear below the POA section on the Convert to Virtual Hearings form

- I affirm that I have the Veteran/Appellant's permission to opt-in to a virtual hearing.

- I affirm that the Veteran/Appellant and POA/Representative will have access to a computer, tablet, or mobile device with a reliable internet connection, a camera, and a microphone on the day of the virtual hearing.

- [X] Selectable “Learn More” link appears at the end affirmation 

- When the “Learn More” link is selected, user is directed to the following location  

- (https://www.bva.va.gov/docs/VirtualHearing_FactSheet.pdf) 

- [X] The two affirmations are required to be selected for representative users to convert hearing to Virtual

1. The user is informed fields are required if user attempts to submit form without selecting

2. The Convert Hearing to Virtual command is disabled until both affirmations are selected

- [X] The affirmations are not required for non-representative users interacting with the form

- [X] The affirmations are hidden for all other non-representative users interacting with the Convert to Virtual form in the Hearings Application

- [X] If checkboxes are not selected, the submission option is disabled.

### Testing Plan
1. Make sure that a VSO user has access to a hearing that can be converted to virtual and is greater than 10 days out:
`h = Hearing.where(disposition: nil).reject{|h| h.virtual? || h.scheduled_for_past?}.first`. Grab the **appeal uuid** with `h.appeal.uuid` and the **hearing uuid** with `h.uuid`
If necessary, use the **hearing uuid** to change the `hearing_day_id` to 130 in the database by running:
`select * from hearings where uuid = 'hearing_uuid_here '`
2. Open up local and sign into Billie VSO
3. navigate to `localhost:3000/queue/appeals/appeal_uuid_goes_here`
4. scroll down to the **Hearing type:** _convert to virtual_ hyperlink
5. On the form, there should be two checkboxes: one affirming the user has the appellant's permission and the other affirming that the appellant has access to a device/internet access for their virtual hearing. The 2nd checkbox has a link to describe what devices are needed for virtual.
![image](https://user-images.githubusercontent.com/99915461/174890819-65fa4bf8-4f4a-40c6-91e3-954b23d4a061.png)

7. The **convert to virtual hearing** button should be disabled while the checkboxes are unchecked.
8. Check one checkbox and then the other. The button should be enabled once both checkboxes are selected. Fill out the form with test data and make sure the form can submit. 
![image](https://user-images.githubusercontent.com/99915461/174890900-b4c5dc35-f256-4b12-9ad6-bfc581429903.png)

10. Sign in as BVASYELLOW to make sure the checkboxes don't appear for non-VSO users. 
11. Go to the Hearing's app. Select a hearing day further than 10 days out to be able to change the hearing to virtual. 
12. Click "Edit Hearing Details" link
13. Click the "Hearing Type" checkbox to convert the hearing from video/travel to virtual
14. Confirm there are no checkboxes at the bottom of the convert to virtual form like the screenshot below:

![image](https://user-images.githubusercontent.com/99915461/174891448-19c5598c-e69c-4685-9a2b-06119023e00e.png)


### User Facing Changes

 BEFORE as VSO
![image](https://user-images.githubusercontent.com/99915461/174891448-19c5598c-e69c-4685-9a2b-06119023e00e.png)


AFTER as VSO
![image](https://user-images.githubusercontent.com/99915461/174890819-65fa4bf8-4f4a-40c6-91e3-954b23d4a061.png)
![image](https://user-images.githubusercontent.com/99915461/174890900-b4c5dc35-f256-4b12-9ad6-bfc581429903.png)
